bee-network/bee-gossip/src/service/host.rs

```diff
@@ -221,13 +221,22 @@ async fn peerstate_checker(shutdown: Shutdown, senders: Senders, peerlist: PeerL
         let num_connected_known = peerlist.filter_count(|info, state| info.relation.is_known() && state.is_connected());
         let num_connected_unknown =
             peerlist.filter_count(|info, state| info.relation.is_unknown() && state.is_connected());
+        let num_connected_discovered =
+            peerlist.filter_count(|info, state| info.relation.is_discovered() && state.is_connected());
 
         info!(
-            "Connected peers: known {}/{} unknown {}.",
-            num_connected_known, num_known, num_connected_unknown,
+            "Connected peers: known {}/{} unknown {}/{} discovered {}/{}.",
+            num_connected_known,
+            num_known,
+            num_connected_unknown,
+            global::max_unknown_peers(),
+            num_connected_discovered,
+            global::max_discovered_peers()
         );
 
-        for (peer_id, info) in peerlist.filter_info(|info, state| info.relation.is_known() && state.is_disconnected()) {
+        for (peer_id, info) in peerlist.filter_info(|info, state| {
+            (info.relation.is_known() || info.relation.is_discovered()) && state.is_disconnected()
```

Comment:
I thought we didn't try to reconnect to discovered peers ?

Reply:
you're right.....I wasn't quite sure at some point whether reconnection should be allowed as long as the peer still belongs to the neighborhood ... good catch

Reply:
Fixed